src/NexusMods.DataModel/Attributes/DiskState.cs

```diff
@@ -0,0 +1,75 @@
+using NexusMods.Abstractions.Games.DTO;
+using NexusMods.Abstractions.Serialization.DataModel.Ids;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.MnemonicDB.Abstractions.Models;
+using NexusMods.Paths;
+
+namespace NexusMods.DataModel.Attributes;
+
+/// <summary>
+/// MnemonicDB attributes for the DiskStateTree registry.
+/// </summary>b
+/// 
+public static class DiskState
+{
+    
+    /// <summary>
+    /// The associated game id.
+    /// </summary>
+    public static readonly Attribute<GameDomain> Game = new("NexusMods.DataModel.DiskStateRegistry/Game", isIndexed: true, noHistory: true);
+    
+    /// <summary>
+    /// The game's root folder
+    /// </summary>
+    public static readonly Attribute<AbsolutePath> Root = new("NexusMods.DataModel.DiskStateRegistry/Root", isIndexed: true, noHistory: true);
+    
+    /// <summary>
+    /// The associated loadout id.
+    /// </summary>
+    public static readonly Attribute<IId> LoadoutRevision = new("NexusMods.DataModel.DiskStateRegistry/LoadoutRevision", isIndexed: true, noHistory: true);
+
+    /// <summary>
+    /// The state of the disk.
+    /// </summary>
+    public static readonly Attribute<NexusMods.Abstractions.DiskState.DiskStateTree> State = new("NexusMods.DataModel.DiskStateRegistry/State", noHistory: true);
```

Comment:
I'm not the biggest fan of uses raw strings like this. Maybe use something like this:

```csharp

    private const string NamespaceString = "NexusMods.DataModel.DiskStateRegistry";

    /// <summary>
    /// The associated game id.
    /// </summary>
    public static readonly Attribute<GameDomain> Game = new($"{NamespaceString}/{nameof(Game)}", isIndexed: true, noHistory: true);

```